test/tasks/suite.yaml

```diff
@@ -0,0 +1,33 @@
+
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: suite
+  namespace: karpenter-tests
+spec:
+  description: |
+    Execute a karpenter test suite
+  params:
+    - name: cluster-name
+      description: Name of the cluster under test.
+    - name: suite-path
+      description: Path to the test suite
+    - name: git-ref
+      description: Git commit, tag, or branch to check out. Requires a corresponding Karpenter snapshot release.
+  workspaces:
+    - name: ws
+  steps:
+
+  - name: checkout-repository
+    image: alpine/git
+    workingDir: $(workspaces.ws.path)
+    script: |
+      git clone https://github.com/aws/karpenter.git .
```

Comment:
we're cloning in both the `before-suite.yaml` and the `suite.yaml`. Do you know how long each clone takes? If we use PVCs for our workspaces instead of emptyDir, we can share the cloned repo and make it so we only do this call once, which I believe would probably speed up every test by a good amount. 

IMO, this logically isolates all setup work from testing better.

You can do this with volumeClaimTemplate: https://tekton.dev/docs/pipelines/workspaces/#example-pipelinerun-definition-using-workspaces

Note that this will also use hostname affinity to schedule the tasks onto the same nodes.

Reply:
I'm really wary of depending on EBS volumes due to the latency, leak potential, failure modes, etc. These clones are quite a bit faster than mounting the volume 

<img width="509" alt="image" src="https://user-images.githubusercontent.com/2585992/179873357-5c021d92-4abd-47b1-bdc7-6103d8db7d69.png">

Reply:
Wow, didn't realize it took only 2 seconds. Fine by me here.